src/Stack/Build/Source.hs

```diff
@@ -405,6 +405,14 @@ loadLocalPackage boptsCli targets (name, (lpv, gpkg)) = do
                 Just STUnknown -> assert False mempty
                 Nothing -> mempty
 
+        filterSkippedComponents = Set.filter (not . (`elem` boptsSkipComponents bopts))
+
+        -- the part for executables currently does nothing as Stack.Build.Execute.exesToBuild
+        -- ignores lpComponents, but once that is changed it should also make this work with executables
+        (exes, tests, benches) = (filterSkippedComponents exeCandidates,
```

Comment:
I added a link to https://github.com/haskell/cabal/issues/2780 as that's the issue which needs to be fixed to make skipping of executables work.